src/include/daos_srv/ras.h

```diff
@@ -0,0 +1,107 @@
+/**
+ * (C) Copyright 2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. 8F-30005.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+/*
+ * RAS event definitions to the used in either data or control planes.
+ */
+
+#ifndef __RAS_H__
+#define __RAS_H__
+
+#define RAS_ID_UNKNOWN_STR "Unknown RAS event"
+#define RAS_SEV_UNKNOWN_STR "Unknown RAS event severity"
+#define RAS_TYPE_UNKNOWN_STR "Unknown RAS event type"
+
+enum ras_event_id {
+	RAS_RANK_FAIL	= 0,
+	RAS_RANK_NO_RESP,
+};
+
+static inline char *
+ras_event_id_enum_to_name(enum ras_event_id id)
+{
+	switch (id) {
+	case RAS_RANK_FAIL:
+		return "daos_rank_failed";
```

Comment:
IMO, these short names should just be string-based representations of the enum names, same as we do for the -DER_ errors. Easier to grep for. I also think that the go-side Stringers should probably use the description instead, as these strings are for humans.

Reply:
I'm assuming you mean the exact equivalent in uppercase, this diverges from the descriptions in the RAS wiki, which I was asked to follow, shall I still go ahead with the change?

Reply:
Ah, I understand now. Hmm. I think this is fine for now. If we're really going to follow the GPFS example, then I think we need to more formally identify these things (e.g. define families of faults by prefix, etc) and also have a way to resolve a string identifier to an enum value. But for the initial stab at this I don't think we need to do all of that. We just need to be prepared for this to continue to evolve.